kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/ResourcesUtilTest.java

```diff
@@ -173,4 +179,78 @@ void toLocalRef() {
+
+        Condition resolvedRefsGen2AndLaterTime = new ConditionBuilder(resolvedRefs)
+                .withLastTransitionTime(now.plusMinutes(1))
+                .withObservedGeneration(2L)
+                .build();
+
+        return List.of(
+                Arguments.of("should add to empty list",
+                        List.of(), resolvedRefs, List.of(resolvedRefs)),
+                Arguments.of("add is idempotent",
+                        List.of(resolvedRefs), resolvedRefs, List.of(resolvedRefs)),
+                Arguments.of("returns totally ordered 1",
+                        List.of(accepted), resolvedRefs, List.of(resolvedRefs, accepted)),
+                Arguments.of("returns totally ordered 2",
+                        List.of(resolvedRefs, accepted), resolvedRefs, List.of(resolvedRefs, accepted)),
+                Arguments.of("returns totally ordered 3",
+                        List.of(accepted, resolvedRefs), resolvedRefs, List.of(resolvedRefs, accepted)),
+                Arguments.of("prefer arg when same observedGeneration 1",
+                        List.of(resolvedRefs), resolvedRefsLaterTime, List.of(resolvedRefsLaterTime)),
+                Arguments.of("prefer arg when same observedGeneration 2",
+                        List.of(resolvedRefsLaterTime), resolvedRefs, List.of(resolvedRefs)),
+                Arguments.of("prefer arg when same observedGeneration 3",
+                        List.of(resolvedRefsGen2AndLaterTime), resolvedRefsGen2, List.of(resolvedRefsGen2)),
+                Arguments.of("prefer arg when same observedGeneration 4",
+                        List.of(resolvedRefsGen2), resolvedRefsGen2AndLaterTime, List.of(resolvedRefsGen2AndLaterTime)),
+                Arguments.of("largest observedGeneration wins 1",
+                        List.of(resolvedRefs), resolvedRefsGen2, List.of(resolvedRefsGen2)),
+                Arguments.of("largest observedGeneration wins 2",
+                        List.of(resolvedRefsLaterTime), resolvedRefsGen2, List.of(resolvedRefsGen2)),
+                Arguments.of("replaces _all_ conditions with same type",
+                        List.of(resolvedRefsLaterTime, resolvedRefs), resolvedRefsGen2, List.of(resolvedRefsGen2)),
+
+                Arguments.of("existing condition with later generation not replaced 1",
+                        List.of(resolvedRefs, resolvedRefsGen2), resolvedRefsLaterTime, List.of(resolvedRefsGen2)),
+
+                Arguments.of("existing condition with later generation not replaced 2",
+                        List.of(resolvedRefsGen2, resolvedRefs), resolvedRefsLaterTime, List.of(resolvedRefsGen2))
+
+        );
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource
+    void maybeAddOrUpdateCondition(String testName, List<Condition> list, Condition condition, List<Condition> expectedResult) {
+        assertThat(ResourcesUtil.maybeAddOrUpdateCondition(list, condition)).isEqualTo(expectedResult);
+    }
```

Comment:
using [`ArgumentSet`](https://junit.org/junit5/docs/5.11.0/api/org.junit.jupiter.params/org/junit/jupiter/params/provider/Arguments.ArgumentSet.html) instead of just `Arguments` means you can avoid the first arg here and the need to add `name = "{0}"`

Reply:
Fixed. Thanks!